mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -127,6 +136,27 @@ abstract class Predictor[
   /** @group setParam */
   def setPredictionCol(value: String): Learner = set(predictionCol, value).asInstanceOf[Learner]
 
+  /**
+   * @group setParam
+   * An overloaded version to support multiple columns for features.
+   *
+   * Please note it will save the features columns names to a new parameter "featuresCols" as
+   * an Array[String].
+   */
+  def setFeaturesCol(value: Array[String]): Learner = {
+    require(isSupportMultiColumnsForFeatures,
+      s"Multiple columns for features is not supported by ${getClass.getSimpleName}")
+    set(featuresCols, value).asInstanceOf[Learner]
+  }
+
+  /**
+   * Children classes should override this to tell whether supporting multiple columns
+   * for features, meaning to fetch features columns names from parameter 'featuresCols'.
+   *
+   * @return Whether this predictor support multiple columns for features
+   */
+  protected def isSupportMultiColumnsForFeatures: Boolean = false
```

Comment:
I do not see the impl that support fit/transform in Predictor/PredictionModel, it seems that in this PR all algs in ML do not impl this new fit/transform with more features cols?